readers/multishard.cc

```diff
@@ -833,20 +834,22 @@ future<> shard_reader_v2::do_fill_buffer() {
             }
             co_await rreader->close();
             std::rethrow_exception(std::move(ex));
-        }).then([this] (reader_and_buffer_fill_result res) {
+        }));
+        {
```

Comment:
Why these sub-scopes for returns (there is one more below)?

Reply:
It's to preserve indentation. Removed in the next patch..